### PR TITLE
SW-5907 Implemented notifications for application submission

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2361** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2360** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2359** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2359?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2358** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2347** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2347?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2348](https://github.com/terraware/terraware-server/pull/2348) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2348?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2361** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2360** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2359** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2359?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2358** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2347** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2347?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2348](https://github.com/terraware/terraware-server/pull/2348) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2348?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2361** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2360** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2359** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2359?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2358** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2347** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2347?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2348](https://github.com/terraware/terraware-server/pull/2348) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2348?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/i18n/Messages.kt [hunk 1]
```diff
@@ -70,27 +70,15 @@ class Messages {
...
-                "Application status $status does not require notifications")
-      }
+  fun applicationSubmittedNotification(
+      organizationName: String,
+      timeSubmitted: Instant
+  ): NotificationMessage =
+      NotificationMessage(
+          getMessage("notification.application.submitted.app.title"),
+          getMessage(
+              "notification.application.submitted.app.body", organizationName, timeSubmitted),
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/i18n/Messages.kt]: The time doesn't appear to be used. The message only has a `{0}` placeholder, no `{1}`.

File: src/main/resources/i18n/Messages_en.properties [hunk 2]
```diff
@@ -120,6 +120,14 @@ notification.accession.dryingEnd.email.subject=An accession has dried.
...
 # Notifications.
 notification.accession.email.linkIntro=Please click the link below to go to your Terraware account where you can view the accession.
+notification.application.submitted.app.title=Application Submitted
+# {0} is Organization name
+notification.application.submitted.app.body=An Application has been submitted for {0}
+# {0} is Organization name
+notification.application.submitted.email.subject=Application submitted for {0}
+# {0} is Organization name, {1} is today's date
+notification.application.submitted.email.body=An Accelerator Application has been submitted for {0} on {1}. You can view the full application in Terraware.
+notification.application.submitted.email.buttonLabel=View Application.
```
Comment (sgrimm) [inline src/main/resources/i18n/Messages_en.properties]: The period at the end doesn't seem right for a button label.

File: src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt [hunk 3]
```diff
@@ -372,6 +373,27 @@ class AppNotificationService(
...
+  fun on(event: ApplicationSubmittedEvent) {
+    systemUser.run {
+      val organizationId = parentStore.getOrganizationId(event.applicationId)
+      if (organizationId == null) {
+        log.error("Organization for application ${event.applicationId} not found")
+        return@run
+      }
+      val organization = organizationStore.fetchOneById(organizationId)
+      val renderMessage = {
+        messages.applicationSubmittedNotification(organization.name, event.time)
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/customer/AppNotificationService.kt]: The time is unused.

**Review by sgrimm — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt [hunk 4]
```diff
@@ -145,10 +153,14 @@ class EmailNotificationService(
     val facilityMonitoringUrl =
         webAppUrls
             .fullFacilityMonitoring(
-                parentStore.getOrganizationId(event.facilityId)!!, event.facilityId)
+                parentStore.getOrganizationId(event.facilityId)!!,
+                event.facilityId,
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt]: Why all the added trailing commas?

**Review by sgrimm — CHANGES_REQUESTED** [review]
EmailNotificationService seems to have a lot of unnecessary changes. I assume it's IntelliJ doing an automated edit or somesuch.

**Comment by tommylau523** [comment]
Not sure how that happened. Maybe something with kfmt. Let me try to revert some of these.

File: src/main/kotlin/com/terraformation/backend/email/model/Models.kt [hunk 5]
```diff
@@ -56,7 +56,9 @@ abstract class EmailTemplateModel(config: TerrawareServerConfig) {
     get() {
       return stringsByLocale.computeIfAbsent(currentLocale()) { _ ->
         FormattingResourceBundleModel(
-            bundle, DefaultObjectWrapperBuilder(Configuration.VERSION_2_3_31).build())
+            bundle,
+            DefaultObjectWrapperBuilder(Configuration.VERSION_2_3_31).build(),
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/email/model/Models.kt:60]: Another stray trailing comma.

**Review by sgrimm — APPROVED** [review]

**Comment by tommylau523** [comment]
### Merge activity

* **Sep 3, 8:41 PM EDT**: @tommylau523 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2361).
* **Sep 3, 9:16 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2361) rebased this pull request as part of a merge.
* **Sep 3, 9:25 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2361).

**Pull request merged**